sdk/metric/manual_reader.go

```diff
@@ -0,0 +1,86 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metric // import "go.opentelemetry.io/otel/sdk/metric"
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"go.opentelemetry.io/otel/sdk/metric/export"
+)
+
+// ManualReader is a a simple Reader that allows an application to
+// read metrics on demand.  It simply stores the Producer interface
+// provided through registration.  Flush and Shutdown are no-ops.
+type ManualReader struct {
+	lock     sync.Mutex
+	producer producer
+	shutdown bool
+}
+
+var _ Reader = &ManualReader{}
+
+// NewManualReader returns an Reader that stores the Producer for
+// manual use and returns a configurable `name` as its String(),
+func NewManualReader() *ManualReader {
```

Comment:
Should this return a `Reader` instead of `*ManualReader`? This would match our `SpanProcessor` constructors and it would mean the `ManualReader` type could be unexported.